template/extra_logging.go

```diff
@@ -1,54 +1,75 @@
 package template
 
 import (
+	"sync"
+
 	"github.com/coveooss/gotemplate/v3/utils"
 	"github.com/coveooss/multilogger"
-	"github.com/sirupsen/logrus"
 )
 
 const (
 	loggingBase = "Logging"
 )
 
 var (
-	// templateLog is the logger used to log message during template processing
-	templateLog *multilogger.MultiLogger = multilogger.New(logrus.InfoLevel, multilogger.DisabledLevel, "", "gotemplate")
+	// TemplateLog is the logger used to log message during template processing
+	TemplateLog = multilogger.New("gotemplate")
 	// InternalLog is application logger used to follow the behaviour of the application
-	InternalLog *multilogger.MultiLogger = multilogger.New(logrus.InfoLevel, multilogger.DisabledLevel, "", "gotemplate")
+	InternalLog = multilogger.New("gotemplate-internal")
 )
 
 var loggingFuncs = dictionary{
-	"trace":    func(args ...interface{}) string { return logBase(templateLog.Trace, args...) },
-	"debug":    func(args ...interface{}) string { return logBase(templateLog.Debug, args...) },
-	"info":     func(args ...interface{}) string { return logBase(templateLog.Info, args...) },
-	"notice":   func(args ...interface{}) string { return logBase(templateLog.Info, args...) },
-	"warning":  func(args ...interface{}) string { return logBase(templateLog.Warning, args...) },
-	"error":    func(args ...interface{}) string { return logBase(templateLog.Error, args...) },
-	"critical": func(args ...interface{}) string { return logBase(templateLog.Fatal, args...) },
-	"fatal":    func(args ...interface{}) string { return logBase(templateLog.Fatal, args...) },
-	"panic":    func(args ...interface{}) string { return logBase(templateLog.Panic, args...) },
+	"trace":   func(args ...interface{}) string { return logBase(TemplateLog.Trace, args...) },
+	"debug":   func(args ...interface{}) string { return logBase(TemplateLog.Debug, args...) },
+	"info":    func(args ...interface{}) string { return logBase(TemplateLog.Info, args...) },
+	"warning": func(args ...interface{}) string { return logBase(TemplateLog.Warning, args...) },
+	"error":   func(args ...interface{}) string { return logBase(TemplateLog.Error, args...) },
+	"fatal":   func(args ...interface{}) string { return logBase(TemplateLog.Fatal, args...) },
+	"panic":   func(args ...interface{}) string { return logBase(TemplateLog.Panic, args...) },
+
+	// Deprecated
+	"notice": func(args ...interface{}) string {
+		noticeWarning.Do(func() {
+			InternalLog.Warning("Please note that notice is deprecated, use info instead.")
+		})
+		return logBase(TemplateLog.Info, args...)
+	},
+	"critical": func(args ...interface{}) string {
+		criticalWarning.Do(func() {
+			InternalLog.Warning("Please note that critical is deprecated, use panic instead.")
+		})
+		return logBase(TemplateLog.Fatal, args...)
```

Comment:
My bad but `critical` shouldn't exit. Maybe `error` would be better. Also fix the warning above